src/Stratis.Bitcoin.Features.BlockStore/BlockStoreQueue.cs

```diff
@@ -213,7 +213,55 @@ public Transaction GetTransactionById(uint256 trxid)
         /// <inheritdoc/>
         public Transaction[] GetTransactionsByIds(uint256[] trxids)
         {
-            throw new NotImplementedException();
+            // Only look for transactions if they're indexed.
+            if (!this.nodeSettings.TxIndex)
+                return null;
+
+            Transaction[] txes = new Transaction[trxids.Length];
+
+            lock (this.blocksCacheLock)
+            {
+                for (int i = 0; i < trxids.Length; i++)
+                {
+                    uint256 txId = trxids[i];
+
+                    foreach (ChainedHeaderBlock chainedHeaderBlock in this.pendingBlocksCache.Values)
```

Comment:
Can't this foreach be replaced with :
```
this.pendingBlocksCache.Values.SelectMany(b=>b.Blocks).SelectMany(b=>b.Transactions)
```

Then you can just FirstOrDefault check if the tx exists?

Reply:
Fixed